### ISSUE TITLE
[Feature Request] Add false positive macro to config

### ISSUE THREAD

**Issue by PEZ** [borkdude/clj-kondo.lsp#2]
clj-kondo complains about macros it doesn't know about, which is fine, because how should it know? 😄 

It would be nice with some help to config away the linting complaints. Maybe as a quick-fix option: ”Lint macro as: ”, or some such. It can be a bit of work for the user to find the config info in the clj-kondo repro, understand it, and configure it correctly, by hand.

**Comment by borkdude** [comment]
Related: https://github.com/borkdude/clj-kondo/issues/543

**Comment by alidcast** [comment]
Ran into this issue trying to create custom React components with `hx` macros (https://github.com/Lokeh/hx/tree/master/src/hx)

the linter is complaining about the function name and arguments, since it's not catching them as being used or passed in, respectively.

in the snippet below, for example, both `App` and `initial-name` param cause lint errors:
```clojure
(defnc App [{:keys [initial-name]}]
  (let [[name] (hooks/useState initial-name)]
    [rn/View [rn/Text {:style {:font-weight "bold"}} name]]))

(defn start
  {:dev/after-load true}
  []
  (expo/render-root (hx/f [App {:initial-name "Hello World!"}])))
```

is there a way to mitigate this?

~currently using this library via vscode-clojure-lint~
^scratch that, looks like this also comes bundled in @PEZ calva, so uninstalling `clojure-lint` got the clj-kondo config overrides working

**Comment by borkdude** [comment]
@alidcastano 

> got the clj-kondo config overrides working

Do I understand correctly that your problem was already fixed and there is no action needed from my side?

**Comment by alidcast** [comment]
well right now i just disabled the `unresolved-symbol` rule which isn't ideal. I'd be preferable to have better support for declaring how macros are initialized/used (or at least disabling warnings per line via comments)

which perhaps there is, and i just need to do more digging into docs - just started using clojure! 😅

**Comment by borkdude** [comment]
@alidcastano Check out the config docs of clj-kondo:

https://github.com/borkdude/clj-kondo/blob/master/doc/config.md

Example:

foo.cljs:
``` clojure
;; I have to guess your namespace form since you didn't provide it:
(ns foo (:require [bar :refer [defnc]]))

(defnc App [{:keys [initial-name]}]
  (let [[name] (hooks/useState initial-name)]
    [rn/View [rn/Text {:style {:font-weight "bold"}} name]]))

(defn start
  {:dev/after-load true}
  []
  (expo/render-root (hx/f [App {:initial-name "Hello World!"}])))
```

No more warnings when linting `bar/defnc` as `clojure.core/defn`:
```
$ clj-kondo --lint /tmp/foo.cljs --config '{:lint-as {bar/defnc clojure.core/defn}}'
linting took 16ms, errors: 0, warnings: 0
```

You can put these rules in `.clj-kondo/config.edn` in your workspace.

**Comment by alidcast** [comment]
oh yeah, I read the config section just wasn't sure what `clojure.core` declaration to alias it to. using `defn` makes sense since they basically have the same API.

thanks for the clarification!

**Comment by borkdude** [comment]
Related: https://github.com/borkdude/clj-kondo/issues/253

**Comment by borkdude** [comment]
I have no plans of implementing this.
